app/main.py

```diff
@@ -1,13 +1,24 @@
-def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+from typing import Any
+
+
+def format_linter_error(error: dict) -> dict[str, str | Any]:
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(err) for err in errors],
```

Comment:
```suggestion
        "errors": [format_linter_error(error) for error in errors],
```
`error` would be better